library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -777,20 +791,13 @@ def CleanCacheRead
 
     def Filter(devices, devregex)
       devices = deep_copy(devices)
-      if devices.nil? || devregex.nil? || devregex == ""
-        return deep_copy(devices)
-      end
+      return devices if devices.nil? || devregex.nil? || devregex == ""
 
-      regex = Ops.add(
-        Ops.add("^(", Ops.get(@DeviceRegex, devregex, devregex)),
-        ")[0-9]*$"
-      )
-      Builtins.y2debug("regex=%1", regex)
-      devices = Builtins.filter(devices) do |file, _devmap|
-        Builtins.regexpmatch(file, regex) == true
-      end
-      Builtins.y2debug("devices=%1", devices)
-      deep_copy(devices)
+      regex = "^(#{@DeviceRegex[devregex] || devregex})[0-9]*$"
+      log.debug("regex=#{regex}")
+      devices.select! { |f, _d| f =~ /#{regex}/ }
```

Comment:
It was ```file``` originally so the natural change is to use ```f``` but we could discuss if maybe is better to use ```n``` for name device or d for device and rename ```_d``` as ```_m```